pkg/storer/uploadstore.go

```diff
@@ -70,15 +79,25 @@ func (db *DB) Upload(ctx context.Context, pin bool, tagID uint64) (PutterSession
 			defer db.events.Trigger(subscribePushEventKey)
 
 			return db.Execute(ctx, func(s internal.Storage) error {
-				return errors.Join(
-					uploadPutter.Close(s, address),
+
+				b, err := s.IndexStore().Batch(ctx)
+				if err != nil {
+					return err
+				}
+
+				err = errors.Join(
+					uploadPutter.Close(s, b, address),
 					func() error {
 						if pinningPutter != nil {
-							return pinningPutter.Close(s, address)
+							return pinningPutter.Close(s, b, address)
 						}
 						return nil
 					}(),
```

Comment:
I'm not sure why this isn't inlined? it's not a blocker since it's an older change but just wondering